contractsV5/KyberTradeLogic.sol

```diff
@@ -473,11 +469,101 @@ contract KyberTradeLogic is KyberHintParser, IKyberTradeLogic, PermissionGroups
         return (reserveArr[bestReserve.index], rates[bestReserve.index], isFeePaying);
     }
 
-    function convertReserveIdToAddresses(bytes8 reserveId)
+    function parseEthToTokenHint(bytes calldata hint)
+        external
+        view
+        returns(
+            TradeType ethToTokenType,
+            bytes8[] memory ethToTokenReserveIds,
+            uint[] memory ethToTokenSplits
+        )
+    {
+        IKyberReserve[] memory ethToTokenAddresses;
+
+        (ethToTokenType, ethToTokenAddresses, ethToTokenSplits) = parseHintE2T(hint);
+
+        bytes8[] memory reserveIds = new bytes8[](ethToTokenAddresses.length);
```

Comment:
why declare an extra paramter and later asign to an unused parameter.
it makes code un necessary long.
longer code is less readable.
also might be more gas to allocate extra variable.
unless compiler is smart enough. which I am not sure.

Reply:
Fixed